src/set_theory/zfc/basic.lean

```diff
@@ -727,6 +727,15 @@ quotient.induction_on y (λ v ⟨a, ha⟩, by { rw (@quotient.sound pSet _ _ _ h
 
 theorem mem_wf : @well_founded Set (∈) := ⟨λ x, induction_on x acc.intro⟩
 
+/-- Recursion on the `∈` relation. -/
+@[elab_as_eliminator]
+def rec_on {p : Set → Sort*} (x) (h : Π x, (Π y ∈ x, p y) → p x) : p x :=
```

Comment:
Why swap `x` and `h` with respect to how they are in `well_founded.fix`?

Reply:
This is the order for `*.rec_on`. If we swap `x` and `h`, then it should be called `Set.rec`.